src/sentry/tasks/symbolication.py

```diff
@@ -99,18 +99,43 @@ def get_native_symbolication_function(data: Any) -> Callable[[Symbolicator, Any]
     return get_native_symbolication_function(data)
 
 
-def get_symbolication_function(
-    data: Any,
-) -> tuple[SymbolicatorPlatform, Callable[[Symbolicator, Any], Any] | None]:
+def get_symbolication_function_for_platform(
+    platform: SymbolicatorPlatform, data: Any
+) -> Callable[[Symbolicator, Any], Any]:
+    """Returns the symbolication function for the given platform
+    and event data."""
+
     from sentry.lang.java.processing import process_jvm_stacktraces
-    from sentry.lang.java.utils import should_use_symbolicator_for_proguard
 
-    if data["platform"] in ("javascript", "node"):
-        return SymbolicatorPlatform.js, process_js_stacktraces
-    elif data["platform"] == "java" and should_use_symbolicator_for_proguard(data.get("project")):
-        return SymbolicatorPlatform.jvm, process_jvm_stacktraces
+    if platform == SymbolicatorPlatform.js:
+        return process_js_stacktraces
+    elif platform == SymbolicatorPlatform.jvm:
+        return process_jvm_stacktraces
     else:
-        return SymbolicatorPlatform.native, get_native_symbolication_function(data)
+        symbolication_function = get_native_symbolication_function(data)
+        # get_native_symbolication_function already returned something in
+        # get_symbolication_platforms
+        assert symbolication_function is not None
+        return symbolication_function
+
+
+def get_symbolication_platforms(data: Any) -> list[SymbolicatorPlatform]:
+    """Returns a list of Symbolicator platforms
+    that apply to this event."""
+
+    from sentry.lang.java.utils import is_jvm_event, should_use_symbolicator_for_proguard
+    from sentry.lang.javascript.utils import is_js_event
+
+    platforms = []
```

Comment:
`find_stacktraces_in_data` is potentially slow, we might want to call it only once here, and then pass the data into the `is_XXX_event` functions.

Reply:
Hm. I see your point, but changing this is honestly kind of annoying.